SeriesGuide/src/main/java/com/battlelancer/seriesguide/ui/dialogs/AddShowDialogFragment.java

```diff
@@ -75,12 +77,21 @@ public static void showAddDialog(SearchResult show, FragmentManager fm) {
         ft.addToBackStack(null);
 
         // Create and show the dialog.
-        DialogFragment newFragment = AddShowDialogFragment.newInstance(show);
+        DialogFragment newFragment = AddShowDialogFragment.newInstance(show, adapter);
         newFragment.show(ft, TAG);
     }
 
-    public static AddShowDialogFragment newInstance(SearchResult show) {
+    /**
+     * Display a dialog which asks if the user wants to add the given show to his show database. If
+     * necessary an AsyncTask will be started which takes care of adding the show.
+     */
+    public static void showAddDialog(SearchResult show, FragmentManager fm) {
+        showAddDialog(show, fm, null);
+    }
+
+    public static AddShowDialogFragment newInstance(SearchResult show, BaseAdapter adapter) {
         AddShowDialogFragment f = new AddShowDialogFragment();
+        f.mAdapterCallback = adapter;
```

Comment:
It's a bad idea to pass objects to fragments other than through the argument `Bundle`, because the system will only use the arguments for recreating the fragment (if it is recreated, user leaves, then returns to app).